File: fizzbuzz/fizz_buzz.ex
defmodule FizzBuzz do

  @spec get(Integer.t) :: String.t
  def get(num) when rem(num, 15) == 0, do: "FizzBuzz"

  def get(num) when rem(num, 3) == 0, do: "Fizz"

  def get(num) when rem(num, 5) == 0, do: "Buzz"

  def get(num), do: num

end

1..100 |> Enum.each(&(IO.puts FizzBuzz.get &1))

File: fizzbuzz/fizz_buzz.erl
-module(fizz_buzz).
-export([run/0, run/1]).

run() -> run(1).
run(N) when N > 100 -> ok;
run(N) ->
    if
        N rem 15 == 0 -> io:write('FizzBuzz');
        N rem 3  == 0 -> io:write('Fizz');
        N rem 5  == 0 -> io:write('Buzz');
        true -> io:write(N)
    end,
    io:nl(),
    run(N + 1).
